packages/daemons/block_generator.go

```diff
@@ -95,15 +96,44 @@ func BlockGenerator(ctx context.Context, d *daemon) error {
 		d.logger.WithFields(log.Fields{"type": consts.DBError, "error": err}).Error("getting all unused transactions")
 		return err
 	}
-
-	// Block generation will be started only if we have transactions
 	if len(trs) == 0 {
 		return nil
 	}
 
+	limits := parser.NewLimits(nil)
+	// Checks preprocessing count limits
+	txList := make([]*model.Transaction, 0, len(trs))
+	for i, txItem := range trs {
+		bufTransaction := bytes.NewBuffer(txItem.Data)
+		p, err := parser.ParseTransaction(bufTransaction)
+		if err != nil {
+			p.ProcessBadTransaction(err)
+			continue
+		}
+		if p.TxSmart != nil {
+			err = limits.CheckLimit(p)
+			if err == parser.ErrLimitStop && i > 0 {
+				model.IncrementAttempt(p.TxHash)
+				break
+			} else if err != nil {
+				if err == parser.ErrLimitSkip {
+					model.IncrementAttempt(p.TxHash)
+				} else {
+					p.ProcessBadTransaction(err)
+				}
+				continue
+			}
+		}
+		txList = append(txList, &trs[i])
+	}
+
+	// Block generation will be started only if we have transactions
```

Comment:
as i remember, in release we returned empty block generation

Reply:
Fixed all comments